vtex/toolbelt
# Adds `support` command

#### What is the purpose of this pull request?
To add the possibility of calling `vtex support <account>`, asking `vtex.support-authority` for a new token with the desired account and a support role. This will only work if there are support apps of __your vendor__ installed in the master workspace of the supported account.

#### What problem is this solving?
This will enable toolbelt to use support apps in production. For more information about support apps, [click here (VTEX devs only)](https://docs.google.com/document/d/19HsGMa9YI_ykMt-0Y1QLFO2zJBpGy52ezCQLPqqvdIg/edit#heading=h.zfxvki4r5d89).

#### How should this be manually tested?
It's easier if you ask me to try it with you! But...

- Make sure you're using **STAGING**.
- Login into account 1, create workspace.
- Clone & `vtex link` `vtex.support-authority`.
- Clone & `vtex link` `vtex.hello-support`.
- `vtex infra install 0.17.1-dev`.
- Login into account 2, create workspace, repeat steps 3-5.
- From account 1, do `vtex support account-2`.
- choose `Support Agent`.
- You'll be logged in in account 2, as a support agent.
- do `vtex ls`! Mess around. You're not allowed to `vtex link` though (Your permissions are defined in `hello-support`).
- When you're done, `vtex switch account-1` again :)

#### Screenshots or example usage
https://asciinema.org/a/KKmTgdqezVByTHNf5MoWFyoDD

#### Types of changes
- [x] Requires change to documentation, which **Hasn't been updated while support apps aren't live yet.**
